main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -465,86 +460,87 @@ object Monomorpher {
 
     case LoweredAst.Expr.Lambda(fparam, exp, tpe, loc) =>
       val (p, env1) = specializeFormalParam(fparam, subst)
-      val e = visitExp(exp, env0 ++ env1, subst)
+      val e = specializeExp(exp, env0 ++ env1, subst)
       MonoAst.Expr.Lambda(p, e, subst(tpe), loc)
 
     case LoweredAst.Expr.ApplyAtomic(op, exps, tpe, eff, loc) =>
-      val es = exps.map(visitExp(_, env0, subst))
+      val es = exps.map(specializeExp(_, env0, subst))
       MonoAst.Expr.ApplyAtomic(op, es, subst(tpe), subst(eff), loc)
 
     case LoweredAst.Expr.ApplyClo(exp1, exp2, tpe, eff, loc) =>
-      val e1 = visitExp(exp1, env0, subst)
-      val e2 = visitExp(exp2, env0, subst)
+      val e1 = specializeExp(exp1, env0, subst)
+      val e2 = specializeExp(exp2, env0, subst)
       MonoAst.Expr.ApplyClo(e1, e2, subst(tpe), subst(eff), loc)
 
     case LoweredAst.Expr.ApplyDef(sym, exps, itpe, tpe, eff, loc) =>
       val it = subst(itpe)
-      val newSym = specializeDefSym(sym, it)
-      val es = exps.map(visitExp(_, env0, subst))
+      val newSym = specializeDefnSym(sym, it)
+      val es = exps.map(specializeExp(_, env0, subst))
       MonoAst.Expr.ApplyDef(newSym, es, it, subst(tpe), subst(eff), loc)
 
     case LoweredAst.Expr.ApplySig(sym, exps, itpe, tpe, eff, loc) =>
       val it = subst(itpe)
       val newSym = specializeSigSym(sym, it)
-      val es = exps.map(visitExp(_, env0, subst))
+      val es = exps.map(specializeExp(_, env0, subst))
       MonoAst.Expr.ApplyDef(newSym, es, it, subst(tpe), subst(eff), loc)
 
     case LoweredAst.Expr.ApplyLocalDef(sym, exps, tpe, eff, loc) =>
       val newSym = env0(sym)
-      val es = exps.map(visitExp(_, env0, subst))
+      val es = exps.map(specializeExp(_, env0, subst))
       val t = subst(tpe)
       val ef = subst(eff)
       MonoAst.Expr.ApplyLocalDef(newSym, es, t, ef, loc)
 
     case LoweredAst.Expr.Let(sym, exp1, exp2, tpe, eff, loc) =>
-      // Generate a fresh symbol for the let-bound variable.
+      // Generate a fresh symbol.
       val freshSym = Symbol.freshVarSym(sym)
       val env1 = env0 + (sym -> freshSym)
-      val e1 = visitExp(exp1, env0, subst)
-      val e2 = visitExp(exp2, env1, subst)
+      val e1 = specializeExp(exp1, env0, subst)
+      val e2 = specializeExp(exp2, env1, subst)
       MonoAst.Expr.Let(freshSym, e1, e2, subst(tpe), subst(eff), loc)
 
     case LoweredAst.Expr.LocalDef(sym, fparams, exp1, exp2, tpe, eff, loc) =>
-      // Generate a fresh symbol for the let-bound variable.
+      // Generate a fresh symbol.
```

Comment:
Loses information.

Reply:
its not really let boud, same coment as the other places

Reply:
Ok, but why even have this comment when the next line is

```scala
val freshSym = Symbol.freshVarSym(sym)
```

The comment previously gave a "why", but now it's just completely redundant IMO.

Reply:
yeah, i agree with that. maybe i should just delete them